### PR TITLE
Pull request for lab 2

### PR DESCRIPTION
**Pull request by ecaterinagherbovei**

### CODE REVIEW

File: PAM_app/app/src/main/java/com/example/pam_app/LoginActivity.kt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    lateinit var nameEditText: EditText
+    lateinit var phoneEditText: EditText
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_login)
+
+        nameEditText = findViewById(R.id.editTextTextPersonName)
+        phoneEditText = findViewById(R.id.editTextPhone)
+        sendButton = findViewById(R.id.sendButton)
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/LoginActivity.kt]: Change init of activity view to android view binding https://developer.android.com/topic/libraries/view-binding

Also change access of view elements to view binding access

File: PAM_app/app/src/main/java/com/example/pam_app/WelcomeActivity.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import java.util.*
+
+class WelcomeActivity : AppCompatActivity() {
+
+    lateinit var main_page_button: Button
+    lateinit var login_button: Button
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_welcome)
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/WelcomeActivity.kt]: Change init of activity view to android view binding https://developer.android.com/topic/libraries/view-binding

File: PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+    internal inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        var image: ImageView = view.findViewById(R.id.imageFood)
+        var name: TextView = view.findViewById(R.id.textFoodName)
+        var price: TextView = view.findViewById(R.id.textPrice)
+    }
+
+    @NonNull
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
+        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.list_item_grid_food, parent, false)
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodAdapter.kt]: Change init of recycler view items to android view binding https://developer.android.com/topic/libraries/view-binding

You can consult example from my sample repository.

File: PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import androidx.annotation.NonNull
+import androidx.recyclerview.widget.RecyclerView
+import com.example.pam_app.R
+
+internal class FoodAdapter(private var foodList: List<FoodModel>) : RecyclerView.Adapter<FoodAdapter.MyViewHolder>() {
+
+    internal inner class MyViewHolder(view: View) : RecyclerView.ViewHolder(view) {
+        var image: ImageView = view.findViewById(R.id.imageFood)
+        var name: TextView = view.findViewById(R.id.textFoodName)
+        var price: TextView = view.findViewById(R.id.textPrice)
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodAdapter.kt]: Change access of view elements to android view binding

File: PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodModel.kt [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    var setGetImage: Int? = image
+        get() = field
+        set(value) {
+            field = value
+        }
+    var setGetName: String? = name
+        get() = field.toString()
+        set(value) {
+            field = value.toString()
+        }
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodModel.kt]: replace self written setter and getters with kotlin properties access https://kotlinlang.org/docs/properties.html

File: PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/RecyclerActivity.kt [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import androidx.recyclerview.widget.RecyclerView
+import com.example.pam_app.R
+
+
+class RecyclerActivity : AppCompatActivity() {
+    private val foodList = ArrayList<FoodModel>()
+    private lateinit var foodAdapter: FoodAdapter
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_recycler)
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/RecyclerActivity.kt]: Change init of activity view to android view binding https://developer.android.com/topic/libraries/view-binding

**Review by ASV44 — APPROVED** [review]

**Review by ASV44 — CHANGES_REQUESTED** [review]
@catea016 Please perform all changes from comments above and ping me when you will finish

**Comment by ecaterinagherbovei** [comment]
@ASV44 eu am introdus schimbarile mentionate mai sus, dar cu setters si getters nu mi se primeste eu am comentat seterele scrise cu kotlin properties, deoarece cand incerc sa accesez functia get() in FoodAdapter, nu o citeste si nu stiu care e cauza

**Comment by ASV44** [comment]
> @ASV44 eu am introdus schimbarile mentionate mai sus, dar cu setters si getters nu mi se primeste eu am comentat seterele scrise cu kotlin properties, deoarece cand incerc sa accesez functia get() in FoodAdapter, nu o citeste si nu stiu care e cauza

@catea016  You have not to access variable by get function. Getter and setters in kotlin are just some decorators and if you add get or set function you can just access property by its name but the get or set function will be invoked.
Just write instead of `getVariableName()`, just `variableName` and everything will work.

**Comment by ecaterinagherbovei** [comment]
@ASV44 thank you! All done

File: PAM_app/app/src/main/java/com/example/pam_app/CategoryActivity.kt [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+            val intent = Intent(this@CategoryActivity, WelcomeActivity::class.java)
+            startActivity(intent)
+        }
+
+        binding.expandAllButton.setOnClickListener {
+            val intent1 = Intent(this@CategoryActivity, com.example.pam_app.foodAndDrinks.RecyclerActivity::class.java)
+            startActivity(intent1)
+        }
+
+
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/CategoryActivity.kt]: Remove empty lines

File: PAM_app/app/src/main/java/com/example/pam_app/WelcomeActivity.kt [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+import com.example.pam_app.databinding.ActivityWelcomeBinding
+import java.text.SimpleDateFormat
+import java.util.*
+
+class WelcomeActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityWelcomeBinding
+
+/*    lateinit var main_page_button: Button
+    lateinit var login_button: Button*/
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/WelcomeActivity.kt]: Remove commented code

File: PAM_app/app/src/main/java/com/example/pam_app/WelcomeActivity.kt [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        val currentDateAndTime: String = simpleDateFormat.format(Date())
+        textView.text = currentDateAndTime
+        //main page button
+       // main_page_button = findViewById(R.id.MainPageButton)
+        binding.MainPageButton.setOnClickListener{
+            val intent = Intent(this@WelcomeActivity, CategoryActivity::class.java)
+            startActivityForResult(intent,1)
+        }
+        //login page button
+       // login_button = findViewById(R.id.loginButton)
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/WelcomeActivity.kt]: Remove commented code

File: PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodModel.kt [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    var setGetPrice: String? = price
+        get() = field.toString()
+        set(value) {
+            field = value.toString()
+        }
```
Comment (ASV44) [inline PAM_app/app/src/main/java/com/example/pam_app/foodAndDrinks/FoodModel.kt]: Change name of variable to image, name and price. There is no point to use such names as `setGet`

**Review by ASV44 — CHANGES_REQUESTED** [review]

**Comment by ecaterinagherbovei** [comment]
@ASV44 done

**Review by ASV44 — APPROVED** [review]
@catea016 PR approved. Feel free to merge it

**Pull request merged**